sql: TestDistSQLDrainingHosts failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/6e0d7081547ee76e52c5f798b78ff739feec2d06

Parameters:
```
TAGS=
GOFLAGS=-race
```

To repro, try:

```
# Don't forget to check out a clean suitable branch and experiment with the
# stress invocation until the desired results present themselves. For example,
# using stress instead of stressrace and passing the '-p' stressflag which
# controls concurrency.
./scripts/gceworker.sh start && ./scripts/gceworker.sh mosh
cd ~/go/src/github.com/cockroachdb/cockroach && \
stdbuf -oL -eL \
make stressrace TESTS=TestDistSQLDrainingHosts PKG=github.com/cockroachdb/cockroach/pkg/sql TESTTIMEOUT=5m STRESSFLAGS='-maxtime 20m -timeout 10m' 2>&1 | tee /tmp/stress.log
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=1208682&tab=buildLog

```
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/stats/automatic_stats.go:247 +0x81e
  github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:200 +0x145

Goroutine 37 (running) created at:
  testing.(*T).Run()
      /usr/local/go/src/testing/testing.go:878 +0x659
  testing.runTests.func1()
      /usr/local/go/src/testing/testing.go:1119 +0xa8
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:827 +0x162
  testing.runTests()
      /usr/local/go/src/testing/testing.go:1117 +0x4ee
  testing.(*M).Run()
      /usr/local/go/src/testing/testing.go:1034 +0x2ee
  github.com/cockroachdb/cockroach/pkg/sql_test.TestMain()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/main_test.go:36 +0x15a
  main.main()
      _testmain.go:598 +0x221
==================
I190329 08:57:31.050778 10197 util/stop/stopper.go:546  quiescing; tasks left:
4      [async] storage.IntentResolver: cleanup txn records
1      [async] transport racer
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
W190329 08:57:31.051299 10169 storage/intentresolver/intent_resolver.go:822  failed to gc transaction record: could not GC completed transaction anchored at /Local/Range/Table/53/1/1/RangeDescriptor: node unavailable; try another peer
I190329 08:57:31.051652 10198 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] storage.IntentResolver: cleanup txn records
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190329 08:57:31.051740 10197 util/stop/stopper.go:546  quiescing; tasks left:
4      [async] storage.IntentResolver: cleanup txn records
1      [async] transport racer
1      [async] intent_resolver_gc_batcher
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190329 08:57:31.052318 10197 util/stop/stopper.go:546  quiescing; tasks left:
4      [async] storage.IntentResolver: cleanup txn records
1      [async] transport racer
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
W190329 08:57:31.052493 10165 storage/intentresolver/intent_resolver.go:822  failed to gc transaction record: could not GC completed transaction anchored at /Local/Range/Table/53/1/1/RangeDescriptor: node unavailable; try another peer
W190329 08:57:31.053078 10124 storage/raft_transport.go:583  [n2] while processing outgoing Raft queue to node 1: EOF:
I190329 08:57:31.054639 10197 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] transport racer
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
W190329 08:57:31.054718 10160 storage/raft_transport.go:583  [n1] while processing outgoing Raft queue to node 2: EOF:
I190329 08:57:31.055309 10198 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190329 08:57:31.058057 10198 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-subscription
1      [async] closedts-rangefeed-subscriber
I190329 08:57:31.060769 10198 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber
I190329 08:57:31.061175 10197 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] transport racer
1      [async] closedts-rangefeed-subscriber
I190329 08:57:31.061477 10197 util/stop/stopper.go:546  quiescing; tasks left:
1      [async] transport racer
I190329 08:57:31.159284 9342 kv/transport_race.go:113  transport race promotion: ran 59 iterations on up to 347 requests
I190329 08:57:31.162094 10186 rpc/nodedialer/nodedialer.go:143  [ct-client] unable to connect to n1: failed to connect to n1 at 127.0.0.1:42819: context canceled
--- FAIL: TestDistSQLDrainingHosts (2.80s)
    testing.go:771: race detected during execution of test

```

--- a/pkg/sql/distsql_physical_planner_test.go
+++ b/pkg/sql/distsql_physical_planner_test.go
@@ -486,7 +486,7 @@ func TestDistSQLDrainingHosts(t *testing.T) {
 		numNodes,
 		base.TestClusterArgs{
 			ReplicationMode: base.ReplicationManual,
-			ServerArgs:      base.TestServerArgs{UseDatabase: "test"},
+			ServerArgs:      base.TestServerArgs{Knobs: base.TestingKnobs{DistSQL: &distsqlrun.TestingKnobs{DrainFast: true}}, UseDatabase: "test"},
 		},
 	)
 	ctx := context.TODO()
@@ -543,7 +543,6 @@ func TestDistSQLDrainingHosts(t *testing.T) {
 	// Drain the second node and expect the query to be planned on only the
 	// first node.
 	distServer := tc.Server(1).DistSQLServer().(*distsqlrun.ServerImpl)
-	distServer.ServerConfig.TestingKnobs.DrainFast = true
 	distServer.Drain(ctx, 0 /* flowDrainWait */)
 
 	expectPlan([][]string{{"https://cockroachdb.github.io/distsqlplan/decode.html#eJyUkDFLxEAUhHt_xTGVwsolV26lWF2TSO7EQoKs2UcIJPvCextQjvx3SbbQE060fDM78w17QmBPhRtIYV-QozYYhRtSZVmk9GDv32Ezgy6MU1zk2qBhIdgTYhd7gsXRvfVUkfMk2wwGnqLr-rV2lG5w8nEXpkFhcBhdULu5hUE5RbspOBDq2YCn-NWv0bUEm8_m7xvu21aodZFlm59PeCifiuNrVT4frm8usnb_YVWkIwelM86l5myuDci3lP5UeZKGHoWbFZPOcs2tgieNyc3TsQ_JWgZ-D-e_hnc_wvV89RkAAP__weakAA=="}})

